cypress/utils/commands.js

```diff
@@ -62,6 +62,80 @@ Cypress.Commands.add('openWorkspaceDashboard', (workspaceName) => {
     .click();
 });
 
+// Command to measure component performance and compare with baseline
+// Exampele - cy.measureComponentPerformance({
+//   page: 'discover',
+//   componentTestId: 'docTable',
+//   eventName: 'onLoadSavedQuery',
+// });
+
+Cypress.Commands.add('measureComponentPerformance', (opts) => {
+  const { page, componentTestId, eventName, isDynamic = false } = opts;
+
+  cy.readFile('cypress/utils/performance_baselines.json').then((baselines) => {
+    const baseline = baselines[page][componentTestId][eventName];
+    const fieldName = `${page}_${componentTestId}_${eventName}`;
+
+    if (!baseline) {
+      cy.log(`No baseline found for component: ${fieldName}`);
+      return;
+    }
+
+    cy.window().then((win) => {
+      win.performance.mark(`${fieldName}_start`);
+
+      cy.get(`[data-test-subj="${componentTestId}"]`).as('component');
+
+      cy.get('@component')
+        .should('exist')
```

Comment:
the `.should('exist')` is not needed, as `cy.get(...)` will fail if it already exists